iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -147,93 +283,119 @@ private Task<MethodResponse> HandleMaxMinReportCommand(MethodRequest request, ob
+                return Task.FromResult(new CommandResponse((int)StatusCode.BadRequest));
             }
         }
 
-        private Task SetDesiredPropertyUpdateCallback(TwinCollection desiredProperties, object userContext)
+        private Task WritablePropertyEventDispatcherAsync(PropertyCollection desiredProperties, object cancellationToken)
         {
-            bool callbackNotInvoked = true;
-
             foreach (KeyValuePair<string, object> propertyUpdate in desiredProperties)
             {
-                string componentName = propertyUpdate.Key;
-                if (_desiredPropertyUpdateCallbacks.ContainsKey(componentName))
+                // The dispatcher key will be either the root-level property name or the component name.
+                string dispatcherKey = propertyUpdate.Key;
+                if (_writablePropertyEventCallbacks.ContainsKey(dispatcherKey))
                 {
-                    _desiredPropertyUpdateCallbacks[componentName]?.Invoke(desiredProperties, componentName);
-                    callbackNotInvoked = false;
+                    return _writablePropertyEventCallbacks[dispatcherKey]?.Invoke(desiredProperties, cancellationToken, dispatcherKey);
                 }
             }
 
-            if (callbackNotInvoked)
+            _logger.LogDebug($"Property: Received a property update that is not implemented.");
+            return Task.CompletedTask;
+        }
+
+        private async Task SendTemperatureRangeAsync(PropertyCollection desiredProperties, object userContext, string dispatcherKey)
+        {
+            string propertyName = dispatcherKey;
+            var cancellationToken = (CancellationToken)userContext;
+
+            // PropertyCollection.Value is now always JObject (since we create PropertyCollection from TwinCollection).
+            // This implementation detail will need to be addressed.
+            string serializedProperties = ((JObject)desiredProperties[propertyName]).ToString();
+            TemperatureRange temeratureRangeDesired = s_payloadConvention.PayloadSerializer.DeserializeToType<TemperatureRange>(serializedProperties);
+
+            var temperatureUpdateResponse = new WritablePropertyResponse(temeratureRangeDesired, s_payloadConvention)
             {
-                _logger.LogDebug($"Property: Received a property update that is not implemented by any associated component.");
-            }
+                AckCode = (int)StatusCode.Completed,
+                AckVersion = desiredProperties.Version,
+                AckDescription = "The operation completed successfully."
+            };
 
-            return Task.CompletedTask;
+            // A property collection with WritablePropertyResponse needs to be always vserialized with NewtonSoft
+            // => we cannot merge it with user defined serialization settings.
+            var propertyPatch = new PropertyCollection();
```

Comment:
TODO: WritablePropertyResponse being added to PropertyCollection needs to be serialized using NewtonSoft (has newtonsoft json property names)

Reply:
Used an abstract base class with virtual overrides. Implemented the Newtonsoft version of this.

Will need to have samples and tutorials on how to extend if using another serializer.